### PR TITLE
Propage call hook helper()

### LINKED ISSUE #1895: call static method in ManageSettings.php
SMF 2.0.8
php 5.4

error: Fatal error: Call to undefined function MyClass::mystaticmethod() in ...\Sources\ManageSettings.php on line 198

find:

``` php
    // Call the right function for this sub-acton.
    $subActions[$_REQUEST['sa']]();
```

replace:

``` php
    $func = strpos($subActions[$_REQUEST['sa']], '::') !== false ? explode('::', $subActions[$_REQUEST['sa']]) : $subActions[$_REQUEST['sa']];
    // Call the right function for this sub-acton.
    if (is_array($func))
        call_user_func($func);
    else
        $func();
```

### PR DESCRIPTION
**Pull request by MissAllSunday**
- Modifies call_hook_helper() to allow executing the given string or return the formated string.
- Uses call_helper() on every subAction call.
- Normalizes $sub_actions and other variations to camelCase $subActions

oh and fixes #1895

### CODE REVIEW

File: Sources/Subs.php [hunk 1]
```diff
@@ -4232,10 +4232,11 @@ function remove_integration_function($hook, $function, $file = '', $object = fal
  * If a method is found, it looks for a "#" which indicates SMF should create a new instance of the given class.
  * Prepare and returns a callable depending on the type of method/function found.
  *
- * @param string $string The string containing a function name
+ * @param string $string The string containing a function name or a static call.
+ * @param boolean $return If true, the function will not call the function/method but instead will return the formatted string.
  * @return string|array Either a string or an array that contains a callable function name or an array with a class and method to call
  */
-function call_hook_helper($string)
+function call_helper($string, $return = false)
```
Comment (live627) [inline Sources/Subs.php:4239]: Should this be renamed to call_user_func_helper?

Comment (MissAllSunday) [inline Sources/Subs.php:4239]: Seems a bit longer and could be confused with call_user_func_array()  In theory, I would like all mod authors to use this on their own modifications.

**Comment by interlab** [comment]
maybe:

find:

``` php
$subActions[$_REQUEST['sa']]();
```

replace:

``` php
call_helper($subActions[$_REQUEST['sa']]);
```

in [smf 2.1, ManageSettings.php, line 131](https://github.com/SimpleMachines/SMF2.1/blob/release-2.1/Sources/ManageSettings.php#L131)

**Comment by live627** [comment]
It's call_helper(), not call_hook_helper()

**Pull request merged**